src/main/cpp/util/path_posix.cc

```diff
@@ -71,12 +69,20 @@ std::string MakeAbsolute(const std::string &path) {
 
 std::string ResolveEnvvars(const std::string &path) {
   std::string result = path;
-  static std::regex env("\\$\\{([^}]+)\\}");
-  std::smatch m;
-  while (std::regex_search(result, m, env)) {
-    const char *value = getenv(m[1].str().c_str());
-    const std::string replacement = std::string(value ? value : "");
-    result.replace(m.position(0), m.length(0), replacement);
+  size_t start = 0;
+  while ((start = result.find("${", start)) != std::string::npos) {
+    // Just match to the next }
+    size_t end = result.find("}", start + 1);
+    if (end == std::string::npos) {
+      // No more variables can exist
+      break;
+    }
+    // Extract the variable name
+    const std::string name = result.substr(start + 2, end - start - 2);
+    // Get the value from the environment
+    const char *value = getenv(name.c_str());
+    result.erase(start, end - start + 1);
+    result.insert(start, value ? value : "");
```

Comment:
I think you want to increment start by end-start+1 here. Otherwise if   FOO='${FOO}' you will loop forever.

I guess the point is that the method comments and docs should specify if there is rescan of the substituted value or not.  I would suggest that no rescan of substituted values has the smallest attack surface.

Reply:
Actually by value.length(), but good point. Also adding a test for this case.